pkg/sac/scope_checker_or.go

```diff
@@ -0,0 +1,212 @@
+package sac
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/sac/effectiveaccessscope"
+)
+
+type orScopeChecker struct {
+	scopeCheckers []ScopeChecker
+}
+
+// NewOrScopeChecker returns a new scope checker wrapping the given scope checkers,
+// each result will be combined in a disjunction (OR).
+func NewOrScopeChecker(scopeCheckers ...ScopeChecker) ScopeChecker {
+	return orScopeChecker{
+		scopeCheckers: scopeCheckers,
+	}
+}
+
+func (s orScopeChecker) Core() ScopeCheckerCore {
+	if len(s.scopeCheckers) != 0 {
+		return s.scopeCheckers[0].Core()
+	}
+	return nil
+}
+
+func (s orScopeChecker) SubScopeChecker(keys ...ScopeKey) ScopeChecker {
+	for i := range s.scopeCheckers {
+		s.scopeCheckers[i] = s.scopeCheckers[i].SubScopeChecker(keys...)
+	}
+	return s
+}
+
+func (s orScopeChecker) PerformChecks(ctx context.Context) error {
+	var performChecksErrs *multierror.Error
+	for _, checker := range s.scopeCheckers {
+		err := checker.PerformChecks(ctx)
+		// Short-circuit on the first non-error perform check.
+		if err != nil {
+			performChecksErrs = multierror.Append(performChecksErrs, err)
+		}
+	}
+	if performChecksErrs == nil {
+		return nil
+	}
+	return performChecksErrs
+}
+
+func (s orScopeChecker) TryAllowed(subScopeKeys ...ScopeKey) TryAllowedResult {
+	result := Deny
+	for _, checker := range s.scopeCheckers {
+		if res := checker.TryAllowed(subScopeKeys...); res == Allow {
+			return res
+		} else if res == Unknown {
+			result = Unknown
+		}
+	}
+	return result
+}
+
+func (s orScopeChecker) Allowed(ctx context.Context, subScopeKeys ...ScopeKey) (bool, error) {
+	var allowedErrs *multierror.Error
+	for _, checker := range s.scopeCheckers {
+		allowed, err := checker.Allowed(ctx, subScopeKeys...)
+		// Short-circuit on the first allowed check result.
+		if err != nil {
+			allowedErrs = multierror.Append(allowedErrs, err)
+		} else if allowed {
+			return allowed, nil
+		}
+	}
+	if allowedErrs != nil {
+		return false, allowedErrs
+	}
+	return false, nil
```

Comment:
`return false, allowedErrs` instead? Same question for similar cases below.

Reply:
Same sentiment as above regarding `allowedErrs` treated as non-nil error.